### PR TITLE
Add k8s troubleshooting docs (hostPort vs hostNetwork)

### PR DESCRIPTION
**Pull request by ishustava**

### CODE REVIEW

File: website/content/docs/troubleshoot/common-errors.mdx
Comment (DerekStrickland) [inline website/content/docs/troubleshoot/common-errors.mdx:168]: Non blocking suggestion - move some of this paragraph before the commands to provide context for what you are about to show.  Something like

### `i/o timeout` or `connection refused` errors 

If you're seeing `i/o timeout` or `connection refused` errors when connecting to the Consul client on the Kubernetes worker, this could be because the CNI (Container Networking Interface) does not [support](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/#support-hostport)
the use of `hostPort`. 

Then show the commands.  Then explain the work around.

**Review by DerekStrickland — APPROVED** [review]
This looks great.  Thanks for adding! Non blocking suggestion in line.

File: website/content/docs/troubleshoot/common-errors.mdx
Comment (thisisnotashwin) [inline website/content/docs/troubleshoot/common-errors.mdx:165]: Does it makes sense to generalize these IP values or mention something along the lines of "where 10.20.62.49" is the cluster IP of the client?

**Review by thisisnotashwin — COMMENTED** [review]
This looks great!! Thanks so much for writing this up. have a question about some of the IPs that have been mentioned in the examples wrt adding some clarification around them but it is excellent documentation otherwise.

File: website/content/docs/troubleshoot/common-errors.mdx
Comment (ishustava) [reply website/content/docs/troubleshoot/common-errors.mdx:168]: will do!

File: website/content/docs/troubleshoot/common-errors.mdx
Comment (ishustava) [reply website/content/docs/troubleshoot/common-errors.mdx:165]: Hmm yeah, I think probably both are good. So maybe we use an IP like `10.0.0.10` or something and also clarify below what that IP is. wdyt?

File: website/content/docs/troubleshoot/common-errors.mdx
Comment (lkysow) [inline website/content/docs/troubleshoot/common-errors.mdx:169]: Could we also say to check that the consul clients are in Running state?

**Review by lkysow — APPROVED** [review]

File: website/content/docs/troubleshoot/common-errors.mdx
Comment (thisisnotashwin) [reply website/content/docs/troubleshoot/common-errors.mdx:165]: LGTM 😄

Comment (thisisnotashwin) [reply website/content/docs/troubleshoot/common-errors.mdx:165]: Approving it assuming those changed will make it and not block this PR 👍

**Review by thisisnotashwin — APPROVED** [review]

**Pull request merged**
